src/redux/missions/missions.js

```diff
@@ -0,0 +1,48 @@
+import { getMissionsFromAPI } from '../../api/spacexdata';
+
+const GET_MISSIONS = 'space-travelers/missions/GET_MISSIONS';
+const JOIN_LEAVE_MISSION = 'space-travelers/missions/JOIN_MISSION';
+const initialState = {};
+
+export const getMissions = (data) => ({
+  type: GET_MISSIONS,
+  payload: data,
+});
+
+export const joinMission = (data) => ({
+  type: JOIN_LEAVE_MISSION,
+  payload: data,
+});
+
+export const getMissionsDispatcher = () => async (dispatch) => {
+  const missions = await getMissionsFromAPI();
+  dispatch(getMissions(missions));
+};
+
+export const joinMissionDispatcher = (id) => (dispatch) => dispatch(joinMission(id));
+export const leaveMissionDispatcher = (id) => (dispatch) => dispatch(joinMission(id));
+
```

Comment:
## Optional
Since those two functions use the same logic. I suggest using just one function if that is possible.

Reply:
Ok thank you, I'll remove it 👍